internal/testutil/helpers/helpers.go

```diff
@@ -7,15 +7,46 @@
 package helpers
 
 import (
+	"context"
 	"fmt"
 	"io/ioutil"
 	"path"
 	"testing"
+	"time"
 
-	"github.com/stretchr/testify/require"
 	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/internal/testutil/require"
 )
 
+// AssertSoon runs the provided callback and fails the passed-in test if the callback
+// does not complete within timeout. The provided callback should respect the
+// passed-in context and cease execution when it has expired.
+func AssertSoon(t testing.TB, callback func(ctx context.Context), timeout time.Duration) {
```

Comment:
I chose to move `assert.Soon` to here in the form of `AssertSoon`. There's no _exact_ analogue in testify. I _could_ refactor all calls to `AssertSoon` to use `assert.Eventually` and get rid of this function if desired.

Reply:
I'm in favor of keeping `Soon` in some form to minimize the scope of this PR and maybe refactoring to use `Eventually` eventually. If that's the direction we want to go, consider adding a comment to this function to indicate that it's deprecated and to use `Eventually` instead.

Reply:
Done. Filed GODRIVER-2667.